timelock-impl/src/main/java/com/palantir/atlasdb/timelock/transaction/timestamp/DelegatingClientAwareManagedTimestampService.java

```diff
@@ -34,7 +36,7 @@
     private static final Logger log = LoggerFactory.getLogger(DelegatingClientAwareManagedTimestampService.class);
 
     @VisibleForTesting
-    static final int NUM_PARTITIONS = 16;
+    static final int NUM_PARTITIONS = TransactionConstants.V2_TRANSACTION_NUM_PARTITIONS;
```

Comment:
Possibly, though it would be a _very_ long way off - this change will typically require a schema migration. It's possible to automate that at some point (basically you will need some way of storing the time at which the number changes and use that to decide how the rows are going to look like) but hard.